File: services/administration-service/src/main/java/sgptt/adminsvc/repository/entity/Person.java
package sgptt.adminsvc.repository.entity;

import java.sql.Timestamp;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Person {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "person_id")
    private Long personId;
    private String name;

    @Column(name = "paternal_surname")
    private String paternalSurname;

    @Column(name = "maternal_surname")
    private String maternalSurname;

    @ManyToOne
    @JoinColumn(name = "role_id")
    private Role role;
    private String email;
    private String password;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "created_at")
    private Timestamp createdAt;

    @Column(name = "is_active")
    private boolean isActive;
}

File: services/auth-service/src/main/kotlin/com/sgpttt/authservice/repository/entity/Person.kt
package com.sgpttt.authservice.repository.entity



import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Temporal
import jakarta.persistence.TemporalType
import java.sql.Timestamp

@Entity
data class Person(
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    @Column(name="person_id")
    val personId: Long,
    val name: String,
    @Column(name="paternal_surname")
    val paternalSurname: String,
    @Column(name="maternal_surname")
    val maternalSurname: String,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "role_id")
    val role: Role,
    val email: String,
    val password: String,
    @Temporal(TemporalType.TIMESTAMP)
    @Column(name="created_at")
    val createdAt: Timestamp,
    @Column(name="is_active")
    val isActive: Boolean
)
